tutorials/networking/port_forwarding_upnp.rst

```diff
@@ -0,0 +1,109 @@
+.. _doc_port_forwarding_upnp:
+
+Port Forwarding & Universal Plug n' Play
+========================================
+
+Why Universal Plug n' Play?
+---------------------------
+
+After you are done testing solo with localhost(127.0.0.1) you want to test on a *real* connection.
+You will be likely overwhelmed with your choices:
+
+1. Host a real server and deploy there (deployment of a build will be a chore unless you learn advanced methods like remote ssh deployment which are included in Godot)
+
+2. Peer-to-Peer connection, straight to a computer. This includes NAT Holepunching, STUN/TURN and other methods which are overwhelming compared to :abbr:`UPNP (Universal Plug n Play)`.
+
+Before any code, let's look into the router setup; Not all Routers support :abbr:`UPNP (Universal Plug n Play)`, nor do all users have :abbr:`UPNP (Universal Plug n Play)` enabled. For this reason it is suggested you use :abbr:`UPNP (Universal Plug n Play)` for testing instead of your final public deployment. Even if you enable your router's :abbr:`UPNP (Universal Plug n Play)` (assuming it is a choice), you need to port-forward the port, because the IP is public but all your router's ports are closed by default.
+
+
+How to port-forward
+-------------------
+
+1. Login to your router (username/password is on the router's bottom where it has barcode tag and original wifi password)
```

Comment:
>We should not provide guidance on how to login to your router. That is way too specific, depends on model and vendor and is also just not our scope. :)

I browsed UPNP setups a bit before I found the 2 videos mentioned in OP whereas I got the final UPNP solution mentioned here. Many people who failed (e.g. https://teddit.net/r/godot/comments/121lul7/godot_4_upnp_invalid_gateway), I bet just failed the port forwarding process. 

As for what you said (quoted above), I tested this succesfully on 2 routers. 1 of them did not have UPNP at all, I just port forwarded and it worked with the UPNP code of this tutorial. And by extension, the sub-tutorial of 'How to port forward' applies to basically all routers, and ensures the user will open his ports instead of trying to just enable his router's UPNP, or be discouraged if he finds his router doesn't have UPNP.

>We should just say something to the effect of "make sure UPnP port forwarding is enabled and supported by your router"

Ideally it would be a collapsable section. I don't know how to do that, does sphinx support such an option?

>Also, I don't get the explanation on creating a port-forward; UPnP is used exactly because it avoids having to do that.

Mentioned far above, tl;dr: Enabling UPnP without opening a port isn't enough for 2/2 routers I used, for external clients to connect.

Reply:
>If you open and forward a port in your router manually, you don't need UPnP. It will just work. 

It didn't in my case. I will make an MRP when I try again, though that will take a while as it is low priority~